engines/wintermute/base/sound/base_sound_manager.cpp

```diff
@@ -100,15 +100,15 @@ BaseSoundBuffer *BaseSoundMgr::addSound(const Common::String &filename, Audio::M
 	BaseSoundBuffer *sound;
 
 	Common::String useFilename = filename;
+	useFilename.toLowercase();
 	// try to switch WAV to OGG file (if available)
-	AnsiString ext = PathUtil::getExtension(filename);
-	if (StringUtil::compareNoCase(ext, "wav")) {
-		AnsiString path = PathUtil::getDirectoryName(filename);
-		AnsiString name = PathUtil::getFileNameWithoutExtension(filename);
-
-		AnsiString newFile = PathUtil::combine(path, name + "ogg");
-		if (BaseFileManager::getEngineInstance()->hasFile(newFile)) {
-			useFilename = newFile;
+	if (useFilename.hasSuffix("wav")) {
+		Common::String oggFilename = useFilename;
+		oggFilename.erase(oggFilename.size() - 3);
+		assert(oggFilename.lastChar() == '.');
+		oggFilename = oggFilename + "ogg";
```

Comment:
Isn't there an append function? I think `s = s + suffix` creates a new string.

Instead of erasing, maybe just set the last 3 chars manually:
```
size = s.size()
s[size - 3] = 'o';
s[size - 2] = 'g';
s[size - 1] = 'g';
```